src/test/java/io/hemin/wien/model/AtomBuiderFactoryTest.java

```diff
@@ -0,0 +1,26 @@
+package io.hemin.wien.model;
+
+import io.hemin.wien.builder.AtomBuilder;
+import io.hemin.wien.builder.PersonBuilder;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class AtomBuiderFactoryTest {
+
+    @Test
+    @DisplayName("should build an Atom model using builder factory methods only")
+    public void testEpisodeBuilderFactory() {
+        PersonBuilder personBuilder = Person.builder()
+            .name("name");
+
+        AtomBuilder atomBuilder = Atom.builder()
+            .addAuthorBuilder(personBuilder);
```

Comment:
The builders from Immutables lib directly add model instances in these cases to the builders, which I find nicer from a producer perspective. This however clashes with our current builder API and usage in the parsers.

Reply:
That would be a very large change, and as you way, contradicts current usage. The current builders model are not eager but rather lazy in their "finalisation". You don't finalise starting from leaf nodes, but start from the root node and it'll finalise everything under itself. This has the advantage of allowing us to refuse building things where mandatory data is missing, and at the same time allows us to build data in any order we wish (useful for example with the bitlove ID, which goes on the enclosure, which may or may not have been built yet)

Reply:
I agree, the current approach has its benefits for us, and I don't want to change that. I just wanted to say that from an API design perspective, I think the direct approach is nicer.

Reply:
Yep. In Kotlin you already have the option to directly construct the data classes with a nice API, it's just Java which is a bit limited.

Reply:
Speaking of, we should consider whether that's something we want to allow in the first place :)

Reply:
That's a very good point. From a Kotlin perspective it's fine. But from a Java perspective, it isn't a good idea to have people create a `new Podcast(title, link, description, pubDate, lastBuildDate, language, generator, copyright, docs, managingEditor, webMaster, image, episodes, iTunes, atom, fyyd, feedpress, googlePlay, categories, podcast)` at all.

Reply:
Unfortunately the only thing we can do is block the ctors from the Kotlin side (by making them `internal`), but Java will ignore `internal` and consider it `public` anyway :(

Reply:
The only other way I can think of is drawing again inspiration from Immutables lib. There, all models are merely defined through interfaces exposing getters only. So the user is forced to use the builders (although this can be changed and customised also). However, w'd the loose the advantage of data classes though then.

Reply:
Actually, we could actually directly use Immutables then. Would at least save us a lot of coding effort. But also makes one of the major Kotlin benefits in this lib obsolete then.

Reply:
I think it's ok to have a good-but-not-perfect Java interop story and prioritise the Kotlin API, since the library itself is written in Kotlin and I'd expect most usages will be from Kotlin projects. It's important to keep compatibility with Java, but IMO shouldn't come at the cost of usability in Kotlin.